scripts/x-data-grid.exports.json

```diff
@@ -449,6 +451,7 @@
   { "name": "GridSearchIcon", "kind": "Variable" },
   { "name": "GridSelectedRowCount", "kind": "Variable" },
   { "name": "GridSeparatorIcon", "kind": "Variable" },
+  { "name": "GridServerSideGroupNode", "kind": "Interface" },
```

Comment:
Would it be possible to replace all "server-side" names with "data source"?

Reply:
Using `dataSource` sounded a bit odd to me in a few places including this. But this could be my personal thinking.

Reply:
I have a small preference for "dataSource" everywhere but I could be convinced otherwise. @cherniavskii Any opinion?

Reply:
I find both options hard to reason about. How about `GridLazyGroupNode`?

Reply:
> How about GridLazyGroupNode?

This is definitely a decent name, however I'm still inclined to use `ServerSide` in this name, for this specific case, one of the reasons would be the key being used in the interface (`hasServerChildren`)
```tsx
interface GridServerSideGroupNode extends GridDataGroupNode {
  /**
   * If true, this node has children on server.
   */
  hasServerChildren: boolean;
  /**
   * The cached path to be passed on as `groupKey` to the server.
   */
  path: string[];
}
```

The other two occasions where it didn't feel great to use `dataSource` and it's still `ServerSide` would be `GridServerSideTreeDataGroupingCell` and `useGridServerSideTreeDataPreProcessors` both of whom I wanted to be consistent with `GridServerSideGroupNode` since all these three variables are inter-related.

We could make all of them to use word `Lazy` though, how do they look?
- `GridLazyGroupNode`
- `GridLazyTreeDataGroupingCell`
- `useGridLazyTreeDataPreProcessors`

Reply:
Alright, using `lazy` everywhere is not great either 😅
Picking out of these three, I would say `dataSource` is the least confusing – it's clear which feature it's related to.
Any objections to use `dataSource` everywhere?

Reply:
Alright then, [updated](https://github.com/mui/mui-x/pull/12317/commits/b3d43eda247dc74ad4293d6b0c9e47f17d946f91) everything to use `DataSource`.